pyformlang/rsa/box.py

```diff
@@ -16,56 +16,57 @@ class Box:
     ----------
     enfa : :class:`~pyformlang.finite_automaton.EpsilonNFA`
         A epsilon nfa
-    label : :class:`~pyformlang.finite_automaton.Symbol`
-        A label for epsilon nfa
+    nonterminal : :class:`~pyformlang.finite_automaton.Symbol`
+        A nonterminal for epsilon nfa
 
     """
 
-    def __init__(self, enfa: EpsilonNFA = None, label: Symbol = None):
-        if enfa is not None:
-            enfa = enfa.minimize()
-        self._dfa = enfa or EpsilonNFA()
-
-        if label is not None:
-            label = to_symbol(label)
-        self._label = label or Symbol("")
-
-    def change_label(self, label: Symbol):
-        """ Set a new label
-
-        Parameters
-        -----------
-        label : :class:`~pyformlang.finite_automaton.Symbol`
-            The new label for automaton
-
-        """
-        self._label = to_symbol(label)
-
-    def change_dfa(self, enfa: EpsilonNFA):
-        """ Set an epsilon finite automaton
-
-        Parameters
-        -----------
-        enfa : :class:`~pyformlang.finite_automaton.EpsilonNFA`
-            The new epsilon finite automaton
-
-        """
-        enfa = enfa.minimize()
+    def __init__(self, enfa: EpsilonNFA, nonterminal: Symbol | str):
```

Comment:
`|` was introduced in Python 3.10. Can you please change it to `Union[Symbol, str]` to be retrocompatible?

Reply:
The problem is a bit everywhere.